core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/UnwrapDateTruncInComparison.java

```diff
@@ -279,8 +279,7 @@ private Object calculateRangeEndInclusive(Object rangeStart, Type type, Supporte
                         case MONTH -> dateTime.plusMonths(1);
                         case YEAR -> dateTime.plusYears(1);
                     };
-                    long endExclusiveMicros = endExclusive.toEpochSecond(ZoneOffset.UTC) * MICROSECONDS_PER_SECOND
-                            + LongMath.divide(endExclusive.getNano(), NANOSECONDS_PER_MICROSECOND, UNNECESSARY);
+                    long endExclusiveMicros = endExclusive.toEpochSecond(ZoneOffset.UTC) * MICROSECONDS_PER_SECOND;
```

Comment:
It would be better to add `verify()` method for ensuring the nanosecond is exactly 0 in my opinion.